packages/datadog-plugin-langchain/src/handlers/default.js

```diff
@@ -0,0 +1,53 @@
+'use strict'
+
+const Sampler = require('../../../dd-trace/src/sampler')
+
+const RE_NEWLINE = /\n/g
+const RE_TAB = /\t/g
+
+// TODO: should probably refactor the OpenAI integration to use a shared LLMTracingPlugin base class
+// This logic isn't particular to LangChain
+class LangChainHandler {
+  constructor (config) {
+    this.config = config
+    this.sampler = new Sampler(config.spanPromptCompletionSampleRate)
+  }
+
+  // no-op for default handler
+  getSpanStartTags (ctx) {}
+
+  // no-op for default handler
+  getSpanEndTags (ctx) {}
+
+  // no-op for default handler
+  extractApiKey (instance) {}
+
+  // no-op for default handler
+  extractProvider (instance) {}
+
+  // no-op for default handler
+  extractModel (instance) {}
+
+  normalize (text) {
+    if (!text) return
+    if (typeof text !== 'string' || !text || (typeof text === 'string' && text.length === 0)) return
+
+    const max = this.config.spanCharLimit
+
+    text = text
+      .replace(RE_NEWLINE, '\\n')
```

Comment:
Do you ever have to deal with Windows newlines? e.g. `\r\n`?

Reply:
Probably not if this is a server response. But if it's data provided from the app or user then it could be a concern.